torch/_inductor/codegen/cpp_gemm_template.py

```diff
@@ -839,14 +803,150 @@ def postprocessor(output):
         template.maybe_append_choice(choices)
         return template
 
-    def render(  # type: ignore[override,return]
+    @staticmethod
+    def get_padded_size(n, block_n, k, should_block_weight):
+        padded_n = get_padded_n(n, block_n)
+        if should_block_weight:
+            new_size = [padded_n // block_n, k, block_n]
+        else:
+            new_size = [k, padded_n]
+        return new_size, padded_n
+
+    @classmethod
+    def prep_weight(cls, inputs, layout, micro_gemm):
+        should_block_weight = (
+            cls == CppGemmTemplate or micro_gemm.get_b_layout() != LayoutType.NORMAL
+        )
+        W = inputs[1]
+        new_inputs = list(inputs)
+        if isinstance(W, ir.IRNode):
+            k, n = W.get_size()[-2:]
+        else:
+            k, n = W.shape[-2:]
+        _, block_n, _ = micro_gemm.register_blocking
+        new_size, padded_n = cls.get_padded_size(n, block_n, k, should_block_weight)
+        padding = padded_n - n
+
+        if isinstance(W, ir.IRNode):
+            if should_block_weight:
+                W = cls.block_weight_irnode(W, new_size, padding)
```

Comment:
Feels consolidation these function of `IRNode` and `Tensor` makes this part of code more readable.